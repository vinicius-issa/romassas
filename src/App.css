h1{
    font-family: Roboto, 'Open Sans', 'Helvetica Neue', sans-serif;
    text-align: center;
    color: #444;
}


.flex-container{
    display: flex !important;
    flex-direction: column;
    width: 100%;
}
