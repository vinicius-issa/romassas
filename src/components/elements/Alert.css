.alert{
    position: fixed;
    margin: -20px 0px;
    align-self: center;
    z-index: 10;
}
