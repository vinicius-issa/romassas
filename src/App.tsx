import React from 'react';
import { Container } from '@material-ui/core';

import Route from './routes'
import './App.css'
import Alert from './components/elements/Alert'



const App: React.FC<{}> = ()=> {
  return (
    <div className="App">
      <Container >
        <Alert />
        <Route/>
      </Container>
    </div>
  );
}


export default App;
