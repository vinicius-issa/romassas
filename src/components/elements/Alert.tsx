import React from 'react';
import { connect } from 'react-redux'
import { Container } from '@material-ui/core'
import { Alert } from '@material-ui/lab'

import { setErrorMessage } from '../../store/actions/errorMessage';
import './Alert.css'

interface Props {
    errorMessage: string;
    setError : any;
}

const AlertEl = ({errorMessage, setError}:Props) => {

    return (<Container id='alert-container' className='flex-container'>
                {errorMessage !== '' &&  
                    <Alert onClose={() => {setError('')}} className="alert" severity="error">{errorMessage}</Alert>
                }
            </Container>)
}

function mapStateToProps(state: any) {
    return {
        errorMessage: state.errorMessage.errorMessage
    }
  }
  
  function mapDispatchToProps(dispatch: any) {
    return {
        setError(error: string) {
            const action = setErrorMessage(error)
            dispatch(action);
        }
    }
}

export default connect(
    mapStateToProps,
    mapDispatchToProps
  )(AlertEl);
  