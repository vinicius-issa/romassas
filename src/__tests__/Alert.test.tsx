import React from 'react';
import App from '../App';
import Enzyme, {  mount} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
import Alert from '../components/elements/Alert';
import {Alert as AlertMaterial} from '@material-ui/lab';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

Enzyme.configure({ adapter: new Adapter() });
const mockStore = configureStore();


describe('Testing App Component', () => {
    it('should no render Alert', () => {
        const store = mockStore({errorMessage:{
            errorMessage: '',
        }});
        const wrapper = mount(
            <Provider store={store}>
                <Alert />
            </Provider>
        );

        expect(wrapper.html()).toMatchSnapshot();
        expect(wrapper.find(AlertMaterial)).toHaveLength(0);

    });

    it('should render Alert with Message = "Error Message"', () => {
        const store = mockStore({errorMessage:{
            errorMessage: 'Error Message',
        }});
        const wrapper = mount(
            <Provider store={store}>
                <Alert />
            </Provider>
        );

        expect(wrapper.html()).toMatchSnapshot();
        expect(wrapper.find(AlertMaterial)).toHaveLength(1);
        expect(wrapper.contains('Error Message')).toEqual(true);
    });
});
