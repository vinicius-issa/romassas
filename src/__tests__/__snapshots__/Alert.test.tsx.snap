// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App Component should no render Alert 1`] = `"<div class=\\"MuiContainer-root flex-container MuiContainer-maxWidthLg\\" id=\\"alert-container\\"></div>"`;

exports[`Testing App Component should render Alert with Message = "Error Message" 1`] = `"<div class=\\"MuiContainer-root flex-container MuiContainer-maxWidthLg\\" id=\\"alert-container\\"><div class=\\"MuiPaper-root MuiAlert-root MuiAlert-standardError alert MuiPaper-elevation0\\" role=\\"alert\\"><div class=\\"MuiAlert-icon\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeInherit\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z\\"></path></svg></div><div class=\\"MuiAlert-message\\">Error Message</div><div class=\\"MuiAlert-action\\"><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"Close\\" title=\\"Close\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeSmall\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button></div></div></div>"`;
